### PR TITLE
Update taggable cache and tests

### PR DESCRIPTION
**Pull request by prisis**

### CODE REVIEW

**Comment by Nyholm** [comment]
Thank you for this PR. 
Have a look at the [Redis adapter](https://github.com/php-cache/redis-adapter) and its Readme. We would like all readmes to follow that template where it's possible.

**Comment by prisis** [comment]
Hey @Nyholm so i should remove the Feature | Supported table?

**Comment by Nyholm** [comment]
Yeah, we have that table on www.php-cache.com instead.

**Comment by prisis** [comment]
Okay, so i can change it on all repositories?

the composer.json info need to be removed too?

And dont merge this, need to squash it

**Comment by Nyholm** [comment]
That would be awesome. 
There is a ticket for that: https://github.com/php-cache/issues/issues/10

Aslo, @aequasi is about to update the URL to the issues, see: https://github.com/php-cache/issues/issues/24 I think that you can use the new URL if you are about to change all repos.

File: README.md [hunk 1]
```diff
@@ -4,8 +4,28 @@
 This is a void implementation of PSR-6. Other names for this adapter could be Blackhole or Null. This
```
Comment (Nyholm) [inline README.md]: Make sure to include the badges

Comment (Nyholm) [inline README.md]: And

> This is a PSR-6 cache implementation for Redis. It is a part of the PHP Cache organisation. To read about features like tagging and hierarchy support please read the shared documentation at www.php-cache.com.

**Comment by prisis** [comment]
Okay, i will do it today :)

File: README.md (hunk 1 above)
Comment (prisis) [inline README.md]: @Nyholm is it okay it they all flat-square and come from shields.io?

**Comment by Nyholm** [comment]
You are awesome. Thanks!

File: README.md (hunk 1 above)
Comment (Nyholm) [inline README.md]: I tested Shields.io just now. It is way slow and seams to fail. 
screen shot 2016-01-25 at 16 53 14

I would suggest to copy from the redis adapter and make a search replace.

``` html
[![Latest Stable Version](https://poser.pugx.org/cache/redis-adapter/v/stable)](https://packagist.org/packages/cache/redis-adapter) [![codecov.io](https://codecov.io/github/php-cache/redis-adapter/coverage.svg?branch=master)](https://codecov.io/github/php-cache/redis-adapter?branch=master) [![Build Status](https://travis-ci.org/php-cache/redis-adapter.svg?branch=master)](https://travis-ci.org/php-cache/redis-adapter) [![Total Downloads](https://poser.pugx.org/cache/redis-adapter/downloads)](https://packagist.org/packages/cache/redis-adapter)  [![Monthly Downloads](https://poser.pugx.org/cache/redis-adapter/d/monthly.png)](https://packagist.org/packages/cache/redis-adapter) [![Software License](https://img.shields.io/badge/license-MIT-brightgreen.svg?style=flat-square)](LICENSE)
```

File: tests/IntegrationPoolTest.php [hunk 2]
```diff
@@ -23,6 +23,11 @@ class IntegrationPoolTest extends BaseTest
         'testHasItem'                        => 'Void adapter does not save,',
         'testDeleteItems'                    => 'Void adapter does not save,',
         'testSave'                           => 'Void adapter does not save,',
+        'testSaveWithoutExpire'              => 'Void adapter does not save,',
+        'testDataTypeFloat'                  => 'Void adapter does only output boolean,',
```
Comment (cryptiklemur) [inline tests/IntegrationPoolTest.php]: `Void adapter only outputs boolean`

Comment (cryptiklemur) [inline tests/IntegrationPoolTest.php]: Same for the other 3

**Comment by prisis** [comment]
@Nyholm  and @aequasi now you can check it again :)

Should i update cache/adapter-common to support only v0.2 or both?

File: tests/IntegrationPoolTest.php [hunk 3]
```diff
@@ -23,6 +23,11 @@ class IntegrationPoolTest extends BaseTest
         'testHasItem'                        => 'Void adapter does not save,',
         'testDeleteItems'                    => 'Void adapter does not save,',
         'testSave'                           => 'Void adapter does not save,',
+        'testSaveWithoutExpire'              => 'Void adapter does not save,',
+        'testDataTypeFloat'                  => 'Void adapter only output boolean,',
```
Comment (cryptiklemur) [inline tests/IntegrationPoolTest.php]: `Void adapter only outputs boolean`

Missed the `s` in `outputs`

Comment (prisis) [inline tests/IntegrationPoolTest.php]: sorry :)

**Comment by Nyholm** [comment]
> Should i update cache/adapter-common to support only v0.2 or both?

If you like to. But there is some work to that. Examples: https://github.com/php-cache/apc-adapter/commit/9200e21dee65e6c3bcda0e0643dc120bc224de72 and https://github.com/php-cache/filesystem-adapter/commit/b3444f999f80039ca141260a928ba0a6f6d09615

File: composer.json [hunk 4]
```diff
@@ -30,12 +30,12 @@
         "php":                  "^5.5|^7.0",
         "psr/cache":            "1.0.0",
         "cache/adapter-common": "^0.1",
-        "cache/taggable-cache": "^0.2"
+        "cache/taggable-cache": "^0.2|^0.3"
```
Comment (Nyholm) [inline composer.json]: You could use tagging `^0.3` and drop 0.2 completely.

**Comment by prisis** [comment]
@Nyholm  done.

File: composer.json [hunk 5]
```diff
@@ -28,14 +28,14 @@
     "require":
     {
         "php":                  "^5.5|^7.0",
-        "psr/cache":            "1.0.0",
-        "cache/adapter-common": "^0.1",
-        "cache/taggable-cache": "^0.2"
+        "psr/cache":            "~1.0",
+        "cache/adapter-common": "^0.2",
```
Comment (Nyholm) [inline composer.json:32]: The BC braking feature in 0.2 is that fetchObjectFormCache must return an array. You should do like this: 

``` php
protected function fetchObjectFromCache($key)
{
    return [false, null];
}
```

**Pull request merged**